tracker/model.go

```diff
@@ -590,3 +591,38 @@ func clamp(a, min, max int) int {
 	}
 	return a
 }
+
+func (m *Model) CollectSendsTo(param Parameter) iter.Seq[sointu.Unit] {
+	return func(yield func(sointu.Unit) bool) {
+		p, ok := param.(NamedParameter)
+		if !ok {
+			return
+		}
+		for _, instr := range m.d.Song.Patch {
+			for _, unit := range instr.Units {
+				if unit.Type != "send" {
+					continue
+				}
+				targetId, ok := unit.Parameters["target"]
+				if !ok || targetId != p.Unit().ID {
+					continue
+				}
+				if !yield(unit) {
+					return
+				}
+			}
+		}
+	}
+}
+
+func (m *Model) InstrumentForUnit(id int) *sointu.Instrument {
```

Comment:
You are returning *sointu.Instrument, but instr is in stack. So, when you return &instr, Go copies the value from stack to heap and returns a pointer to that, which means unnecessary heap allocation.

If you really would want to return a pointer to the instrument, &m.d.Song.Patch[...] would do the trick, but we really don't want to expose the m.d.Song directly to the GUI, because the whole point of Model package was to make sure gioui does not directly change the data without telling the Model.

Elsewhere, I even make a deep copy of the instrument to not expose the data to the gioui:

https://github.com/vsariola/sointu/blob/daf7fb1519856a0ab279815f19d4907bf48902e3/tracker/model.go#L407

Both places would benefit from figuring out how we efficiently tell the gioui side of things what sends point where, without exposing entire instruments to the gioui.

Reply:
that part is now unified in the new cache structure, no Copy() anymore